vendor/symfony/lib/routing/sfRoute.class.php

```diff
@@ -283,7 +283,7 @@ protected function generateWithTokens($parameters)
         case 'variable':
           if (!$optional || !isset($this->defaults[$token[3]]) || $parameters[$token[3]] != $this->defaults[$token[3]])
           {
-            $url[] = urlencode($parameters[$token[3]]);
+            $url[] = QubitRoute::urlencode3986($parameters[$token[3]]);
```

Comment:
We should try to not make changes in the Symfony classes when possible. I'd overwrite the entire function in the QubitRoute class. I'm not sure if all the other QubitRoute classes extend QubitRoute or sfRoute, so you may need to do the same in others or add an intermediate class.